src/toxcore/messenger/packet/nickname.rs

```diff
@@ -0,0 +1,88 @@
+/*! Nickname struct.
+*/
+
+use std::str;
+use nom::rest;
+
+use toxcore::binary_io::*;
+
+/// Maximum size in bytes of nickname string of nickname packet
+const MAX_NICKNAME_DATA_SIZE: usize = 128;
+
+/** Nickname is a struct that holds string of my nickname.
+
+This packet is used to transmit sender's nickname to a friend.
+Every time a friend become online or my nickname is changed,
+this packet is sent to the friend or to all friends of mine.
+
+Serialized form:
+
+Length    | Content
+--------- | ------
+`1`       | `0x30`
+`0..128`  | UTF8 byte string
+
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Nickname {
+    nickname: String,
+}
+
+impl FromBytes for Nickname {
+    named!(from_bytes<Nickname>, do_parse!(
+        tag!("\x30") >>
+        nickname: map_res!(rest, str::from_utf8) >>
+        verify!(value!(nickname), |nickname: &str| nickname.len() <= MAX_NICKNAME_DATA_SIZE) >>
```

Comment:
I'd check the length first, only after convert to string, like:
```rust
map_res!(
    verify!(rest, |nickname: &[u8]| nickname.len() <= MAX_NICKNAME_DATA_SIZE),
    str::from_utf8
)
```

Reply:
done